orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -360,8 +357,12 @@ def _create_training_session(self):
                          "CUDAExecutionProvider"])
             providers.append("CPUExecutionProvider")
             if self._use_external_cuda_allocator:
-                provider_options = [{"device_id": str(self._device.index), "cuda_external_alloc": str(
-                    self._torch_alloc), "cuda_external_free": str(self._torch_free)}, {}]
+                if self.is_rocm_pytorch:
+                    provider_options = [{"device_id": str(self._device.index), "rocm_external_alloc": str(
+                        self._torch_alloc), "rocm_external_free": str(self._torch_free)}, {}]
+                else:
+                    provider_options = [{"device_id": str(self._device.index), "cuda_external_alloc": str(
```

Comment:
How about renaming rocm_external_alloc/cuda_external_alloc to gpu_external_alloc and rocm_external_free/cuda_external_free to gpu_external_free?